From 9441134d67f0c2386d7fcdbbea3274d13f0f5a02 Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:52:01 -0400
Subject: [tools] perf: Make perf_event__synthesize_mmap_events global

Message-id: <1408953210-26343-234-git-send-email-jolsa@redhat.com>
Patchwork-id: 89040
O-Subject: [PATCH RHEL7.1 BZ1133083 233/322] perf tools: Make perf_event__synthesize_mmap_events global
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit a18382b68f8bf1a8d43e3cb08b3479cb768913ea
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Jan 7 13:47:20 2014 +0100

description
===========
Making perf_event__synthesize_mmap_events global, it will be used in
following patch from test code.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/event.c b/tools/perf/util/event.c
index 45a76c6..1fc1c2f 100644
--- a/tools/perf/util/event.c
+++ b/tools/perf/util/event.c
@@ -175,12 +175,12 @@ out:
  return tgid;
 }
 
-static int perf_event__synthesize_mmap_events(struct perf_tool *tool,
-           union perf_event *event,
-           pid_t pid, pid_t tgid,
-           perf_event__handler_t process,
-           struct machine *machine,
-           bool mmap_data)
+int perf_event__synthesize_mmap_events(struct perf_tool *tool,
+           union perf_event *event,
+           pid_t pid, pid_t tgid,
+           perf_event__handler_t process,
+           struct machine *machine,
+           bool mmap_data)
 {
  char filename[PATH_MAX];
  FILE *fp;
diff --git a/tools/perf/util/event.h b/tools/perf/util/event.h
index 30fec99..faf6e21 100644
--- a/tools/perf/util/event.h
+++ b/tools/perf/util/event.h
@@ -266,6 +266,13 @@ int perf_event__synthesize_sample(union perf_event *event, u64 type,
       const struct perf_sample *sample,
       bool swapped);
 
+int perf_event__synthesize_mmap_events(struct perf_tool *tool,
+           union perf_event *event,
+           pid_t pid, pid_t tgid,
+           perf_event__handler_t process,
+           struct machine *machine,
+           bool mmap_data);
+
 size_t perf_event__fprintf_comm(union perf_event *event, FILE *fp);
 size_t perf_event__fprintf_mmap(union perf_event *event, FILE *fp);
 size_t perf_event__fprintf_mmap2(union perf_event *event, FILE *fp);
-- 
1.7.1